### PR TITLE
Using one list database account action instead of two

### LINKED ISSUE #1967: [DocDB] List command requires a resource group parameter
It would be great if we could remove the resource group as a required parameter for the `az documentdb list` command, in order to be consistent with the general behavior of `list` commands in the CLI.

### PR DESCRIPTION
**Pull request by dmakwana**
Fixes #1967. Taking the resource-group-name parameter as optional so which can allow the -list action to be used for both resource-group-based listings or subscription-based listings

### CODE REVIEW

**Comment by dmakwana** [comment]
Issue: #1967

**Comment by tjprescott** [comment]
Please add: "Fixes #1967" to the PR description. This will allow GitHub to automatically close the issue when this PR is merged.

File: src/command_modules/azure-cli-documentdb/azure/cli/command_modules/documentdb/_params.py [hunk 1]
```diff
@@ -62,3 +62,5 @@ def validate_ip_range_filter(ns):
 register_cli_argument('documentdb update', 'max_staleness_prefix', help="when used with Bounded Staleness consistency, this value represents the number of stale requests tolerated. Accepted range for this value is 1 - 2,147,483,647", type=int)
 register_cli_argument('documentdb update', 'max_interval', help="when used with Bounded Staleness consistency, this value represents the time amount of staleness (in seconds) tolerated. Accepted range for this value is 1 - 100", type=int)
 register_cli_argument('documentdb update', 'ip_range_filter', validator=validate_ip_range_filter, help="firewall support. Specifies the set of IP addresses or IP address ranges in CIDR form to be included as the allowed list of client IPs for a given database account. IP addresses/ranges must be comma separated and must not contain any spaces", nargs='+')
+
+register_cli_argument('documentdb list', 'resource_group_name', help="name of the resource group")
```
Comment (tjprescott) [inline src/command_modules/azure-cli-documentdb/azure/cli/command_modules/documentdb/_params.py]: You shouldn't need this entry. 'resource_group_name' is globally aliased and, in general, you shouldn't mess with that.

File: src/command_modules/azure-cli-documentdb/azure/cli/command_modules/documentdb/custom.py [hunk 2]
```diff
@@ -108,3 +108,14 @@ def cli_documentdb_update(client,
...
     docdb_account = client.database_accounts.get(resource_group_name, account_name) # Workaround
     return docdb_account
+
+
+def cli_documentdb_list(client,
+                        resource_group_name=None):
+    """Lists all Azure DocumentDB database accounts within a given resource group or subscription.
+    """
+
+    if resource_group_name is None:
```
Comment (tjprescott) [inline src/command_modules/azure-cli-documentdb/azure/cli/command_modules/documentdb/custom.py]: You can just say `if not resource_group_name:`  Empty string isn't a valid resource group name. Or, better still:

```Python
if resource_group_name:
  # call list_by_resource_group
else:
  # call list all
```

**Comment by dmakwana** [comment]
Fixes #1967

**Pull request merged**